src/FSLibrary/MaxTPSTest.fs

```diff
@@ -0,0 +1,237 @@
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+module MaxTPSTest
+
+// This module provides a configurable max TPS test to use in missions
+
+open Logging
+open StellarCoreHTTP
+open StellarCorePeer
+open StellarCoreSet
+open StellarFormation
+open StellarMissionContext
+open StellarNetworkData
+open StellarStatefulSets
+open StellarSupercluster
+
+// Get the maximum value from a distribution
+let private maxDistributionValue (distribution: (int * int) list) = fst (List.maxBy fst distribution)
+
+// Upgrade max tx size to 2x the maximum possible from the distributions in
+// `context`
+let private upgradeSorobanTxLimits (context: MissionContext) (formation: StellarFormation) (coreSetList: CoreSet list) =
+    formation.SetupUpgradeContract coreSetList.Head
+
+    let multiplier = 2
+    let instructions = int64 ((maxDistributionValue context.instructionsDistribution) * multiplier)
+    let txBytes = (maxDistributionValue context.totalKiloBytesDistribution) * multiplier * 1024
+    let entries = (maxDistributionValue context.dataEntriesDistribution) * multiplier
+    let txSizeBytes = (maxDistributionValue context.txSizeBytesDistribution) * multiplier
+    let wasmBytes = (maxDistributionValue context.wasmBytesDistribution) * multiplier
+
+    formation.DeployUpgradeEntriesAndArm
+        coreSetList
+        { LoadGen.GetDefault() with
+              mode = CreateSorobanUpgrade
+              txMaxInstructions = Some instructions
+              txMaxReadBytes = Some txBytes
+              txMaxWriteBytes = Some(max txBytes wasmBytes)
+              txMaxReadLedgerEntries = Some entries
+              txMaxWriteLedgerEntries = Some entries
+              txMaxSizeBytes = Some(max txSizeBytes wasmBytes)
+              maxContractSizeBytes = Some(maxDistributionValue context.wasmBytesDistribution * multiplier)
+              // Memory limit must be reasonably high
+              txMemoryLimit = Some 200000000
+              minSorobanPercentSuccess = Some 100 }
```

Comment:
this isn't a network config, so seems redundant here

Reply:
Sorry, with github's formatting it's hard to tell which part you're talking about. Are you talking about the whole `LoadGen` config here? Or some of the specific options?

Reply:
Ah, got it. Yes, it's redundant because of the `WaitForTxMaxInstructions` call. I removed it.